server/sockets/cohort.js

```diff
@@ -1,136 +1,201 @@
-  responseWithCohortDetails
+  responseWithCohortDetails,
+  responseWithListsMetaData
 } = require('../common/utils');
 const {
   cohortChannel,
   descriptionLockId,
   emitCohortMetaData,
+  getListsDataByViewers,
+  getListIdsByViewers,
   handleLocks,
   nameLockId,
-  removeCohort
+  listChannel
 } = require('./helpers');
 const { isDefined } = require('../common/utils');
+const List = require('../models/list.model');
 
-const addCohortMember = (socket, clients) =>
-  socket.on(CohortActionTypes.ADD_MEMBER_SUCCESS, data => {
-    const { cohortId } = data;
+const addMember = (io, allCohortsClients, dashboardClients) => data => {
+  const {
+    cohortId,
+    member: { _id: userId }
+  } = data;
 
-    socket.broadcast
-      .to(cohortChannel(cohortId))
-      .emit(CohortActionTypes.ADD_MEMBER_SUCCESS, data);
+  io.sockets
+    .to(cohortChannel(cohortId))
+    .emit(CohortActionTypes.ADD_MEMBER_SUCCESS, data);
+
+  if (allCohortsClients.size > 0) {
+    emitCohortMetaData(cohortId, allCohortsClients, io);
+  }
+
+  List.find(
+    {
+      cohortId,
+      type: ListType.SHARED
+    },
+    '_id created_at cohortId name description items favIds type'
+  )
+    .lean()
+    .exec()
+    .then(docs => {
+      if (docs && docs.length > 0) {
+        const lists = responseWithListsMetaData(docs, userId);
+        const sharedListIds = lists.map(list => list._id.toString());
+        const { member } = data;
+        const viewer = {
+          ...member,
+          isMember: false,
+          isViewer: true
+        };
+
+        sharedListIds.forEach(listId => {
+          io.sockets
+            .to(listChannel(listId))
+            .emit(ListActionTypes.ADD_VIEWER_SUCCESS, {
+              listId,
+              ...viewer
+            });
+        });
 
-    if (clients.size > 0) {
-      emitCohortMetaData(cohortId, clients, socket);
-    }
-  });
+        const id = userId.toString();
 
-const leaveCohort = (socket, clients) =>
-  socket.on(CohortActionTypes.LEAVE_SUCCESS, data => {
-    const { cohortId } = data;
+        if (dashboardClients.has(id)) {
+          const { socketId } = dashboardClients.get(id);
+          const dataMap = _keyBy(lists, '_id');
 
-    socket.broadcast
-      .to(cohortChannel(cohortId))
-      .emit(CohortActionTypes.REMOVE_MEMBER_SUCCESS, data);
+          io.sockets
+            .to(socketId)
+            .emit(ListActionTypes.FETCH_META_DATA_SUCCESS, dataMap);
+        }
+      }
+    });
+};
 
-    if (clients.size > 0) {
-      emitCohortMetaData(cohortId, clients, socket);
-    }
-  });
+const leaveCohort = (io, allCohortsClients) => data => {
+  const { cohortId, userId } = data;
 
-const addOwnerRoleInCohort = (socket, clients) => {
-  socket.on(CohortActionTypes.ADD_OWNER_ROLE_SUCCESS, data => {
-    const { cohortId, userId } = data;
+  io.sockets
+    .to(cohortChannel(cohortId))
+    .emit(CohortActionTypes.REMOVE_MEMBER_SUCCESS, data);
+
+  if (allCohortsClients.size > 0) {
```

Comment:
part of this code was he same  so we made helper for that (`emitCohortMetaData`):
```
  if (allCohortsClients.size > 0) {
    emitCohortMetaData(cohortId, allCohortsClients, io);
  }
```
the rest of the code is different, but I made a helper for changing role in cohort because of code was exactly the same